airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -62,10 +60,31 @@ def path(self, **kwargs) -> str:
         return f"repos/{self.repository}/{self.name}"
 
     def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        response_data = response.json()
-        if response_data and len(response_data) == self.page_size:
-            self._page += 1
-            return {"page": self._page}
+        links = response.links
+        if "next" in links:
+            next_link = links["next"]["url"]
+            parsed_link = parse.urlparse(next_link)
+            page = dict(parse.parse_qsl(parsed_link.query)).get("page")
+            return {"page": page}
```

Comment:
what about if `page` is not in the parsed query params? should this be `{"page": page} if page else None`?

Reply:
it is not possible, that is why it is next URL, the pagination is well defined in the API